flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -106,6 +106,108 @@ static bool isAbsent(const fir::ExtendedValue &exv) {
   return !fir::getBase(exv);
 }
 
+/// Process calls to Minloc or Maxloc intrinsic functions
+template <typename FN, typename FD>
+static fir::ExtendedValue
+genMinMaxloc(FN func, FD funcDim, mlir::Type resultType,
+             Fortran::lower::FirOpBuilder &builder, mlir::Location loc,
+             Fortran::lower::StatementContext *stmtCtx, 
+             const char *errMsg, llvm::ArrayRef<fir::ExtendedValue> args) {
+  assert(args.size() == 5);
+
+  // Handle required array argument
+  auto array = builder.createBox(loc, args[0]);
+
+  fir::BoxValue arryTmp = builder.createBox(loc, args[0]);
+  int rank = arryTmp.rank();
+  assert(rank >= 1);
+
+  // Handle optional mask argument
+  auto mask = isAbsent(args[2]) ?
+              builder.create<fir::AbsentOp>( 
+              loc, fir::BoxType::get(builder.getI1Type())) :
+              builder.createBox(loc, args[2]);
+
+  // Handle optional kind argument
+  auto kind = isAbsent(args[3]) ?
+              builder.createIntegerConstant(loc, builder.getIndexType(),
+                                   builder.getKindMap().defaultIntegerKind()) :
+              fir::getBase(args[3]); 
+
+  // Handle optional back argument
+  auto back = isAbsent(args[4]) ?
+              builder.createBool(loc, false) :
+              fir::getBase(args[4]);
+
+  bool absentDim = isAbsent(args[1]);
+
+  if (!absentDim && rank == 1) {
+    // use a scalar result descriptor with Min/MaxlocDim().
```

Comment:
This probably needs some more explanation as to why the case of DIM is present and ARRAY rank is 1 is special.